enhancements/ocp-coreos-layering.md

```diff
@@ -103,23 +109,117 @@ A future enhancement will help automate the above.  For example, we may support
+
+They can copy and extend these and build them using any container build system they like.
+The administrator can also use non-Dockerfile build systems such as Tekton or buildah.
+
+The result will be a container image that can be passed via an API to clusters, in a similar way as way user workloads (pods).
+
+### API Extensions ([MCO-280](https://issues.redhat.com/browse/MCO-280))
+
+Note we will continue to support MachineConfig.  The key goal is to allow the user to override the rhel-coreos base image.
+
+#### Current default proposal: Configure via imagestream
+
+This is currently prototyped in the https://github.com/openshift/machine-config-operator/tree/layering branch.  Here there’s an imagestream per MachineConfigPool, and the administrator can specify an image which overrides the default rhel-coreos base image.
+
+##### User workflow in layering model
+Using layering model, users will be able to build, test and use a custom image based on their requirements.
+The user experience is described in the sub-sections below:
+
+###### Create Customer Base Image
+- First of all the user will create a Customer Base Image. This image can run on compatible OCP clusters such as HyperShift, Heterogeneous cluster, Single Node OpenShift.
+- Create a Dockerfile that uses compatible rhcos image and add the custom content (e.g. rpm, files) on top of it:
+  ```
+  FROM registry.redhat.io/rhel-coreos:4.11
+  RUN cd /etc/yum.repos.d/ && \
+    curl -LO <user_yum_repo> && \
+    rpm-ostree install <pkg> && \
+    rpm-ostree cleanup -m && \
+    systemctl enable <services> && \
+    ostree container commit
+  ADD <conf_file> /etc/<conf_file>
+  ```
+- Build the image using any image build tool (e.g. podman) and push it to the registry (e.g. quay)
+- Generated Customer Base Image can be manually tested or integrated with a CI workflow to run desired internal tests. For example, this image can be run through a security scanner the same as any other image.
+- Once all the tests on the image are green, it can be tagged for use in a OCP cluster, like:
+  ```
+  oc tag --source=docker quay.io/examplecorp/custom-rhel-coreos:4.11 machine-config-operator/coreos-external:latest
+  ```
+- Once the image has been tagged,  some verification will be performed in the cluster to ensure that rhcos version present in the Customer Base Image is compatible with the cluster and feedback will be provided accordingly.
+    - **On compatibility check passed**: “quay.io/examplecorp/custom-rhel-coreos:4.11 successfully tagged”
+    - **On compatibility check failed**: “failed to tag quay.io/examplecorp/custom-rhel-coreos:4.11: rhcos version 89.2423423 not compatible with OCP 4.11.5, expected: rhcos version 89.43920432”
+
+###### Create Final Pool Images
+When the Customer Base Image has been successfully tagged into cluster, a final image will be created for each pool.
```

Comment:
The previous section talks about tagging images, but not tagging them "into a cluster". Is that something different?